test/functional/tests/ci/test_ci_write_mode_read_only.py

```diff
@@ -0,0 +1,187 @@
+#
+# Copyright(c) 2022 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+import pytest
+import time
+
+from api.cas import casadm
+from api.cas.cli import casadm_bin
+from core.test_run import TestRun
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from test_utils.size import Unit, Size
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.nand, DiskType.optane]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.CI()
+def test_ci_write_mode():
+    """
+    title: Verification test for write mode: read only
+    description: Verify if write mode: read only, works as expected and cache only reads
+    and do not cache writes
+    pass criteria:
+    - cache does not cache write
+    """
+
+    with TestRun.step("Prepare devices"):
+        cache_device = TestRun.disks["cache"]
+        core_device = TestRun.disks["core"]
+
+        cache_device.create_partitions([Size(1024, Unit.MebiByte)])
+        core_device.create_partitions([Size(2048, Unit.MebiByte)])
+
+        cache_device = cache_device.partitions[0]
+        core_device = core_device.partitions[0]
+
+    with TestRun.step("Test read from cache"):
+        count = 100
+        block_size = 4096
+        iostat_block_size = 1024
+        modes = ["wt", "wa"]
```

Comment:
I'd prefer using existing enums to strings, which might be longer, but perhaps more maintainable, and if you use casadm wrapper for --set-cache-mode, you can use the enums as args